### PR TITLE
Add support for democracynow.org

### PR DESCRIPTION
**Pull request by atomicdryad**
Supports downloading clips or entire shows. Subtitle support

### CODE REVIEW

**Comment by atomicdryad** [comment]
broken merge, resubmitting

**Comment by atomicdryad** [comment]
Closed the wrong PR

File: youtube_dl/extractor/democracynow.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        formats = []
+        subtitles = {}
+        for key in ('caption_file', '.......'):
+            # ....... = pending vtt support that doesn't clobber srt 'chapter_file':
+            url = js.get(key, '')
+            if url == '' or url is None:
+                continue
+            if not re.match(r'^https?://', url):
+                url = base_host + url
+            ext = re.search(r'\.([^\.]+)$', url).group(1)
```
Comment (yan12125) [inline youtube_dl/extractor/democracynow.py]: Seems this line just extracts the file extension from the url. There's no need to explicitly provide ```ext``` if file extensions can be determined from URLs.

File: youtube_dl/extractor/democracynow.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        re_desc = re.search(r'<meta property=.og:description. content=(["\'])(.+?)\1', webpage, re.DOTALL)
+        description = re_desc.group(2) if re_desc else ''
+
+        jstr = self._search_regex(r'({.+?"related_video_xml".+?})', webpage, 'json', default=None)
+        js = self._parse_json(jstr, display_id)
+        video_id = None
+        formats = []
+        subtitles = {}
+        for key in ('caption_file', '.......'):
+            # ....... = pending vtt support that doesn't clobber srt 'chapter_file':
```
Comment (yan12125) [inline youtube_dl/extractor/democracynow.py]: Not quite understand. Why not add vtt files to ```subtitles```, too?

File: youtube_dl/extractor/democracynow.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                url = url + 'start=' + str(js.get('start'))
+            formats.append({
+                'format_id': purl.group('dir'),
+                'ext': purl.group('ext'),
+                'url': url,
+            })
+        self._sort_formats(formats)
+        ret = {
+            'id': video_id,
+            'title': js.get('title'),
```
Comment (yan12125) [inline youtube_dl/extractor/democracynow.py]: If 'title' is not in ```js```, there should be a fallback title.

**Comment by yan12125** [comment]
Thanks for your work @atomicdryad. I've made some changes, especially removing ```start=X``` and ```end=Y``` parameters from the final URLs.

And, would you like to be listed in [AUTHORS](https://github.com/rg3/youtube-dl/blob/master/AUTHORS)? If so, under what name?

**Pull request merged**